### PR TITLE
[imaging_uploader] fix forbidden file types that could be uploaded

### PR DESCRIPTION
**Pull request by cmadjar**
## Brief summary of changes

This fixes the issue of any file type being able to be uploaded when the 'Phantom Scans' was set to 'Yes'. 

#### Testing instructions (if applicable)

1. make sure you cannot upload file types different than `.zip`, `.tgz` or `.tar.gz` both when 'Phantom Scans' is set to 'Yes' and 'No'

#### Links to related tickets (GitHub, Redmine, ...)

* #5526

### CODE REVIEW

**Comment by cmadjar** [comment]
@kongtiaowang could you review this PR? It should fix the issue you reported :).

**Comment by johnsaigle** [comment]
@cmadjar If possible could you add back-end validation here too? This should give a 400 response when the files are submitted to the back-end.

A person will still be able to upload other file types by submitting a raw POST request and ignoring the front-end.

**Comment by cmadjar** [comment]
@johnsaigle the check of the file type is already done in imaging_upload.class.inc on line 381 (when the file gets validated). The `isValid` function is called by the `_saveFile` function which is called by the `_process` function so I guess everything is fine, no?

**Comment by gdevenyi** [comment]
Beyond checking filenames, I think validation should be done by testing the file magic numbers before doing any further processing, either calling "file" and parsing the output, or perhaps there's a module for PHP.

**Comment by cmadjar** [comment]
@gdevenyi it looks like this is already done after having checked the filename. Line 397 of `imaging_uploader.class.inc` calls the function `isCompressed` on the file which is checking the mime_content_type of the file.

It does not look like this is checked on the JS side though. Not sure if this is an issue...

**Comment by johnsaigle** [comment]
@cmadjar My understanding from the ticket was that a file upload of forbidden types successfully occurred. That would mean that the check on the back-end is not called or is broken.

I haven't done any testing for this issue though.

**Comment by cmadjar** [comment]
Ha! Thank you @johnsaigle for the reminder!! Indeed, on the PHP side, the checking was also done only on the non-phantom scans... Moved the file type check out of the `if (isPhantom=='N')` statement since it should apply to all uploaded files :).

**Comment by johnsaigle** [comment]
I tested the front-end checks but not the back-end.

File: modules/imaging_uploader/jsx/UploadForm.js [hunk 1]
```diff
@@ -81,6 +81,34 @@ class UploadForm extends Component {
     }
 
     const fileName = data.mriFile.name;
+    // Make sure file is of type .zip|.tgz|.tar.gz format
+    const properExt = new RegExp('.(zip|tgz|tar.gz)$');
```
Comment (driusan) [inline modules/imaging_uploader/jsx/UploadForm.js]: The "."s in this regex aren't escaped, "." means "any character" in a regular expression not a literal "."

Comment (johnsaigle) [reply modules/imaging_uploader/jsx/UploadForm.js]: Is this fixed by a backslash?

```suggestion
    const properExt = new RegExp('\.(zip|tgz|tar.gz)$');
```

I'm not sure good with JS

File: modules/imaging_uploader/jsx/UploadForm.js [hunk 2]
```diff
@@ -81,6 +81,34 @@ class UploadForm extends Component {
...
+    if (!fileName.match(properExt)) {
+      swal({
+        title: 'Invalid extension for the uploaded file!',
+        text: 'Filename extension does not match .zip, .tgz or .tar.gz ',
+        type: 'error',
+        confirmButtonText: 'OK',
+      });
+
+      let errorMessage = {
+        mriFile: 'The file ' + fileName + ' is not of type .tgz, .tar.gz or .zip.',
```
Comment (johnsaigle) [inline modules/imaging_uploader/jsx/UploadForm.js]:
```suggestion
        mriFile: 'The file ' + fileName + ' must be of type .tgz, .tar.gz or .zip.',
```

Not a big deal but I think this is a bit more instructive

File: modules/imaging_uploader/jsx/UploadForm.js (hunk 1 above)
Comment (driusan) [reply modules/imaging_uploader/jsx/UploadForm.js]: yes, but you're missing the one in tar.gz and I think it needs to be retested

Comment (driusan) [reply modules/imaging_uploader/jsx/UploadForm.js]: @cmadjar

Comment (cmadjar) [reply modules/imaging_uploader/jsx/UploadForm.js]: right. I saw that and meant to correct it but I guess I was distracted ;). Done in the last commit

**Comment by driusan** [comment]
@johnsaigle can you retest just in case there's some interaction with string and regex escaping?

**Comment by johnsaigle** [comment]
front-end check is working

**Pull request merged**